temporal-sdk/src/main/java/io/temporal/internal/worker/Poller.java

```diff
@@ -232,27 +235,45 @@ public void run() {
           suspender.await();
         }
 
-        if (pollExecutor.isShutdown()) {
+        if (shouldTerminate()) {
           return;
         }
+
         task.run();
         pollBackoffThrottler.success();
+      } catch (InterruptedException e) {
+        // we restore the flag here, so it can be checked and processed (with exit) in finally
+        Thread.currentThread().interrupt();
```

Comment:
Should we also do `pollBackoffThrottler.failure()` to avoid leaking?

Reply:
I would say no. InterruptedException is not really something that should trigger a backoff, it's not really a "failure", it's just a sign that we should terminate the thread because something is canceled or shutting down. 
But I don't think it's very important or critical here, we can add it if you think it looks better this way.